src/map.rs

```diff
@@ -1167,10 +1167,7 @@ macro_rules! iterator_impl {
                                                 write_ptr = write_ptr.offset(1);
                                             }
                                             External(key, ref $($mut_)* value) => {
-                                                self.remaining_max -= 1;
-                                                if self.remaining_min > 0 {
-                                                    self.remaining_min -= 1;
-                                                }
+                                                self.remaining -= 1;
```

Comment:
There shouldn't be, since this iterator now tracks the exact number of elements remaining.